src/config.c

```diff
@@ -378,75 +510,92 @@ static long long getNumeric(const char *name, void *privdata)
         return c->connection_timeout;
-    } else if (strcasecmp(name, CONF_JOIN_TIMEOUT) == 0) {
+    } else if (strcasecmp(name, conf_join_timeout) == 0) {
         return c->join_timeout;
-    } else if (strcasecmp(name, CONF_RESPONSE_TIMEOUT) == 0) {
+    } else if (strcasecmp(name, conf_response_timeout) == 0) {
         return c->response_timeout;
-    } else if (strcasecmp(name, CONF_PROXY_RESPONSE_TIMEOUT) == 0) {
+    } else if (strcasecmp(name, conf_proxy_response_timeout) == 0) {
         return c->proxy_response_timeout;
-    } else if (strcasecmp(name, CONF_RECONNECT_INTERVAL) == 0) {
+    } else if (strcasecmp(name, conf_reconnect_interval) == 0) {
         return c->reconnect_interval;
-    } else if (strcasecmp(name, CONF_LOG_MAX_FILE_SIZE) == 0) {
+    } else if (strcasecmp(name, conf_log_max_file_size) == 0) {
         return (long long) c->log_max_file_size;
-    } else if (strcasecmp(name, CONF_LOG_MAX_CACHE_SIZE) == 0) {
+    } else if (strcasecmp(name, conf_log_max_cache_size) == 0) {
         return (long long) c->log_max_cache_size;
-    } else if (strcasecmp(name, CONF_SHARDGROUP_UPDATE_INTERVAL) == 0) {
+    } else if (strcasecmp(name, conf_shardgroup_update_interval) == 0) {
         return c->shardgroup_update_interval;
-    } else if (strcasecmp(name, CONF_MAX_APPEND_REQ_IN_FLIGHT) == 0) {
+    } else if (strcasecmp(name, conf_max_append_req_in_flight) == 0) {
         return c->max_appendentries_inflight;
     }
 
     return REDISMODULE_ERR;
 }
 
 
-static int setNumeric(const char *name, long long val, void *priv, RedisModuleString **err)
+static int setNumeric(const char *name, long long val, void *priv,
+                      RedisModuleString **err)
 {
     RedisRaftCtx *rr = &redis_raft;
     RedisRaftConfig *c = priv;
 
     if (rr->state != REDIS_RAFT_UNINITIALIZED) {
-        if (!strcasecmp(name, CONF_ID)) {
+        if (!strcasecmp(name, conf_id)) {
             *err = RedisModule_CreateString(NULL, err_init, strlen(err_init));
             return REDISMODULE_ERR;
         }
     }
 
-    if (strcasecmp(name, CONF_ID) == 0) {
+    if (strcasecmp(name, conf_id) == 0) {
         c->id = (raft_node_id_t) val;
-    } else if (strcasecmp(name, CONF_PERIODIC_INTERVAL) == 0) {
+    } else if (strcasecmp(name, conf_periodic_interval) == 0) {
         c->periodic_interval = (int) val;
-    } else if (strcasecmp(name, CONF_REQUEST_TIMEOUT) == 0) {
-        c->request_timeout = (int) val;
-    } else if (strcasecmp(name, CONF_ELECTION_TIMEOUT) == 0) {
-        c->election_timeout = (int) val;
-    } else if (strcasecmp(name, CONF_CONNECTION_TIMEOUT) == 0) {
+    } else if (strcasecmp(name, conf_request_timeout) == 0) {
+        int p = (int) val;
+
+        if (rr->state == REDIS_RAFT_UP) {
+            if (raft_config(rr->raft, 1, RAFT_CONFIG_REQUEST_TIMEOUT, p) != 0) {
+                *err = RedisModule_CreateStringPrintf(NULL, "Failed to configure raft");
+                return REDISMODULE_ERR;
```

Comment:
maybe add more detailed about the specific failed configuration

Reply:
I think only thing we can mention is it is an internal error. This error check is just a precaution if in future something changes in libraft and we forget to update this part in RedisRaft. I've changed error message to "Failed to configure request timeout: internal error".